wxGrid and wxHTML fixes

Kees-van-der-Oord-Nikon opened the pull request:
fixes for two flaws:
- in grid.cpp fix mix up of wxALIGN_CENTER, wxALIGN_CENTER_VERTICAL and wxALIGN_CENTER_HORIZONTAL
- in htmlpars.cpp added support for HTML charset attribute

Kees-van-der-Oord-Nikon committed:
Update grid.cpp

fixed mix up of vertical and horizontal alignment

Kees-van-der-Oord-Nikon committed:
Update htmlpars.cpp

support HTML 5 charset attribute

Kees-van-der-Oord-Nikon committed:
Merge branch 'master' of https://github.com/Kees-van-der-Oord-Nikon/wxWidgets

vadz commented:
Thanks for the fixes, but could you please explain how to reproduce the bug fixed by the first commit (in `wxGrid`)? It doesn't seem that obvious to me which alignment constant should be used here, so I'd really like to test the fix before merging it.

TIA!

Kees-van-der-Oord-Nikon committed:
Update grid.cpp

add support for wxGrid::DrawTextRectangle alignment arguments values  wxALIGN_CENTRE_HORIZONTAL and wxALIGN_CENTER_VERTICAL

Kees-van-der-Oord-Nikon commented:
Dear Vadz,

I fixed this in a rather large project that include the xlnt library to load .xlsx files, it would take me considerable time to make a minimal example. I hope the arguments below are sufficient for you to understand the details.

When calling wxGrid::DrawTextRectangle(dc,text,wxALIGN_CENTER_HORIZONTAL,wxALIGN_CENTER_VERTICAL,0),
the current implementation does not align the text correctly. The first pull request fixed that but it broke compatibility with the situation that the function is called like this:
wxGrid::DrawTextRectangle(dc,text,wxALIGN_CENTER,wxALIGN_CENTER,0);

I just pushed a second pull request that restores the original case line and only adds case lines to support wxALIGN_CENTER_HORIZONTAL and wxALIGN_CENTER_VERTICAL.

WR,

Kees

vadz commented:
The problem is that it's really not clear whether the parameter should be `wxALIGN_CENTRE` or `wxALIGN_CENTRE_{HORIZONTAL,VERTICAL}`. Accepting both of them should be fine, but perhaps it would be even better to change the caller using the latter constants to use `wxALIGN_CENTRE` instead?

How/when exactly does this function end up being called with `wxALIGN_CENTRE_{HORIZONTAL,VERTICAL}` in your application?
